cmd/service/mesosturbo_service.go

```diff
@@ -0,0 +1,168 @@
+func NewMesosTurboService() *MesosTurboService {
+	s := MesosTurboService{}
+	return &s
+}
+
+// AddFlags adds flags for a specific VMTServer to the specified FlagSet
+func (s *MesosTurboService) AddFlags(fs *pflag.FlagSet) {
+	fs.IntVar(&s.LogVersion, "v", s.LogVersion, "Log level")
+	fs.StringVar(&s.MesosMasterConfig, "mesosconfig", s.MesosMasterConfig, "Path to the mesos config file.")
+	fs.StringVar(&s.TurboCommConfig, "turboconfig", s.TurboCommConfig, "Path to the turbo config flag.")
+
+	fs.StringVar(&s.Master, "mesostype", s.Master, "Mesos Master Type 'Apache Mesos'|'Mesosphere DCOS'")
+	fs.StringVar(&s.MasterIP, "masterip", s.MasterIP, "IP for the Mesos Master")
+	fs.StringVar(&s.MasterPort, "masterport", s.MasterPort, "Port for the Mesos Master")
+	fs.StringVar(&s.MasterUsername, "masteruser", s.MasterUsername, "User for the Mesos Master")
+	fs.StringVar(&s.MasterPassword, "masterpwd", s.MasterPassword, "Password for the Mesos Master")
+
+	fs.StringVar(&s.TurboServerUrl, "turboserverurl", s.TurboServerUrl, "Url for Turbo Server")
+	fs.StringVar(&s.OpsManagerUsername, "opsmanagerusername", s.OpsManagerUsername, "Username for Ops Manager")
+	fs.StringVar(&s.OpsManagerPassword, "opsmanagerpassword", s.OpsManagerPassword, "Password for Ops Manager")
+}
+
+// Run runs the specified VMTServer.  This should never exit.
+func (s *MesosTurboService) Run(_ []string) error {
+
+	probeCategory := string(conf.CloudNative)
+
+	targetConf :=  s.MesosMasterConfig
+	turboCommConf := s.TurboCommConfig
+
+	// ----------- Mesos Target Config
+	var mesosTargetConf *conf.MesosTargetConf
+	var mesosConfErr error
+	if targetConf != "" {
+		mesosTargetConf, mesosConfErr = conf.NewMesosTargetConf(targetConf)
+	} else {
+		var master conf.MesosMasterType
+		if s.Master == string(conf.Apache) {
+			master = conf.Apache
+		} else if s.Master == string(conf.DCOS){
+			master = conf.DCOS
+		}
+		if (master == "") {
+			mesosConfErr = fmt.Errorf("Mesos Master Type is required")
+		}
+		if (s.MasterIP == "") {
+			mesosConfErr = fmt.Errorf("Mesos Master IP is required")
+		}
+
+		mesosTargetConf = &conf.MesosTargetConf{
+			Master: master,
+			MasterIP: s.MasterIP,
+			MasterPort: s.MasterPort,
+			MasterUsername: s.MasterUsername,
+			MasterPassword: s.MasterPassword,
+		}
+	}
+	if mesosTargetConf == nil || mesosConfErr != nil {
+		glog.Errorf("Cannot start Mesos TAP service, invalid target config : %s\n", mesosConfErr.Error())
+		os.Exit(1)
+	}
+
+	mesosMasterType := mesosTargetConf.Master
+
+	// ---------- Turbo server config
+	var turboCommConfigData *service.TurboCommunicationConfig
+	var turboConfErr error
+	if turboCommConf != "" {
+		turboCommConfigData, turboConfErr = service.ParseTurboCommunicationConfig(turboCommConf)
+	} else {
+		wsConfig := mediationcontainer.WebSocketConfig {
+			LocalAddress:    "http://127.0.0.1",
+			WebSocketUsername:   "vmtRemoteMediation",
+			WebSocketPassword: "vmtRemoteMediation",
+			WebSocketPath: "/vmturbo/remoteMediation",
```

Comment:
We'd better not set any default value here. Actually, the default value for web socket config is set during validation, [here](https://github.com/turbonomic/turbo-go-sdk/blob/37cb7b066b91c00759fcccaea2ce72ca02e503b1/pkg/mediationcontainer/mediation_communication_config.go#L40).